commcare_connect/opportunity/helpers.py

```diff
@@ -430,119 +464,153 @@ def get_opportunity_delivery_progress(opp_id):
-    aggregates["accrued_since_yesterday"] = accrued_since_yesterday
 
-    return aggregates
+    return annotated_opportunity.first()
 
 
 def get_opportunity_worker_progress(opp_id):
-    today = now().date()
-    yesterday = today - timedelta(days=1)
-    opportunity = (
-        Opportunity.objects.filter(id=opp_id).values("start_date", "end_date", "total_budget", "currency").first()
-    )
-
-    aggregates = Opportunity.objects.filter(id=opp_id).aggregate(
-        total_deliveries=Count("opportunityaccess__completedwork", distinct=True),
-        approved_deliveries=Count(
-            "opportunityaccess__completedwork",
-            filter=Q(opportunityaccess__completedwork__status=CompletedWorkStatus.approved),
-            distinct=True,
-        ),
-        rejected_deliveries=Count(
-            "opportunityaccess__completedwork",
-            filter=Q(opportunityaccess__completedwork__status=CompletedWorkStatus.rejected),
-            distinct=True,
-        ),
-        total_accrued=OpportunityAnnotations.total_accrued(),
-        total_paid=OpportunityAnnotations.total_paid(),
-        visits_since_yesterday=Count("uservisit", filter=Q(uservisit__visit_date__gte=yesterday), distinct=True),
+    return (
+        Opportunity.objects.filter(id=opp_id)
+        .annotate(
+            total_deliveries=get_deliveries_count_subquery(),
+            approved_deliveries=get_deliveries_count_subquery(CompletedWorkStatus.approved),
+            rejected_deliveries=get_deliveries_count_subquery(CompletedWorkStatus.rejected),
+            total_accrued=total_accrued_sq(),
+            total_paid=total_paid_sq(),
+            visits_since_yesterday=deliveries_from_yesterday_sq(),
+        )
+        .first()
     )
-    aggregates.update(opportunity)
-
-    return aggregates
 
 
 def get_opportunity_funnel_progress(opp_id):
-    completed_user_ids = (
-        CompletedModule.objects.filter(opportunity=OuterRef("pk"))
-        .values("user")
+    started_deliveries_sq = (
+        UserVisit.objects.filter(opportunity_id=OuterRef("pk"))
+        .values("opportunity_id")
+        .annotate(count=Count("user_id", distinct=True))
+        .values("count")
+    )
+
+    return (
+        Opportunity.objects.filter(id=opp_id)
         .annotate(
-            completed_modules=Count("module", distinct=True),
-            total_modules=Subquery(
-                LearnModule.objects.filter(app=OuterRef("opportunity__learn_app"))
-                .values("app")
-                .annotate(count=Count("id"))
-                .values("count")[:1]
+            workers_invited=OpportunityAnnotations.workers_invited(),
+            pending_invites=OpportunityAnnotations.pending_invites(),
+            started_learning_count=OpportunityAnnotations.started_learning(),
+            claimed_job=Count("opportunityaccess__opportunityclaim", distinct=True),
+            started_deliveries=Coalesce(Subquery(started_deliveries_sq, output_field=IntegerField()), 0),
```

Comment:
nit, move the Subquery and Coalesce wrapping to the the `started_deliveries_sq` variable declaration